sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -377,6 +378,17 @@ abstract class JdbcDialect extends Serializable with Logging {
   }
 
   private[jdbc] class JDBCSQLBuilder extends V2ExpressionSQLBuilder {
+    def inputToCaseWhenSQL(input: Expression): String =
+      "CASE WHEN " + inputToSQL(input) + " THEN 1 ELSE 0 END"
+
+    def inputToIntSQL(expr: Expression): String = {
+      expr match {
+        case p: Predicate if p.name() != "ALWAYS_TRUE" && p.name() != "ALWAYS_FALSE" =>
```

Comment:
where? do not understand the question...
there are more dbs that do not support booleans, so sql server is not the only one. it is for now though, but we should make it future-proof

Reply:
The code here assumes we don't need to turn `"ALWAYS_TRUE"` into int, we need to prove that this assumption is true. Can you give me the code pointers?

Reply:
```
override def compileValue(value: Any): Any = value match {
    case booleanValue: Boolean => if (booleanValue) 1 else 0
    case other => super.compileValue(other)
  }
```

in MsSqlServerDialect

Reply:
so back to my first question: is this specific to sqlserver? If yes, shall we document it so that the callers of this method should do the same thing as `MsSqlServerDialect`